### ISSUE TITLE
Consider providing String constants in IanaLinkRelations

### ISSUE THREAD

**Issue by vpavic** [spring-projects/spring-hateoas#1216]
Currently, it's not possible to reuse `IanaLinkRelations` in use cases such as `Relation#collectionRelation` due to limitation of annotation attributes that must be compile time constants.

It would be nice if `IanaLinkRelations` could be extended to provide `String` constant values in similar manner like in `MediaTypes`.

If this change is acceptable, I'd be glad to provide a PR.

**Comment by gregturn** [comment]
That's a good idea.

Indeed if you extract the string values into constants adjacent to the `LinkRelation` type, I would certainly support that.

```java
public static final String ABOUT_VALUE = "about";

public static final LinkRelation ABOUT = LinkRelation.of(ABOUT_VALUE);
```

Like that?

**Comment by vpavic** [comment]
Yep, that's exactly what I had in mind.

So, should I proceed with putting together a PR to address this?

**Comment by gregturn** [comment]
Please do. You can ask me to then review it if you wish. 

PS a test case leveraging it the way you described would look great.

**Comment by odrotbohm** [comment]
Can you elaborate on your use case? `LinkRelation` exists to avoid dealing with Strings in the first place. Mostly because `LinkRelation`s need to adhere to a dedicated format.

**Comment by gregturn** [comment]
Annotations (@Resource) would be able to reference a static constant string instead of plugging in a hand written string. 

Is that not what is stated in the initial description?

**Comment by odrotbohm** [comment]
Ah, I didn't realize `Relation` was pointing to our annotation. Maybe I would just not have expected that those are referred to for customizations. But yeah, of course, makes sense to be consistent then.

**Comment by vpavic** [comment]
I've just opened #1219 to address this.

**Comment by gregturn** [comment]
Merged and backported to 1.0.

Thanks @vpavic!
